src/DistributedArrays.jl

```diff
@@ -1253,4 +1256,84 @@ function Ac_mul_B(A::DMatrix, B::AbstractMatrix)
     return Ac_mul_B!(one(T), A, B, zero(T), C)
 end
 
+
+# Sorting a DVector using samplesort
+
+function sample_n_setup_ref(d::DVector, sample_size)
+    lp = localpart(d)
+    llp = length(lp)
+    sample_size = llp > sample_size ? sample_size : llp
+    sample = lp[rand(1:llp, sample_size)]
```

Comment:
All I know about sample sort is from http://parallelcomp.uw.hu/ch09lev1sec5.html and it doesn't seem that you follow the implementation here. I'd expect a sort and an equally spaces sample taken at this step.